contracts/Strategy.sol

```diff
@@ -170,79 +176,68 @@ contract Strategy is BaseStrategy {
 
         // 2 forms of profit. Incentivized rewards (BAL+other) and pool fees (want)
         collectTradingFees();
-        sellRewards();
+        // this would allow finer control over harvesting to get credits in without selling
+        if (toggles.doSellRewards) {
+            _sellRewards();
+        }
 
         uint256 afterWant = balanceOfWant();
 
         _profit = afterWant.sub(beforeWant);
         if (_profit > _loss) {
             _profit = _profit.sub(_loss);
+            _debtPayment += _loss;
             _loss = 0;
         } else {
             _loss = _loss.sub(_profit);
+            _debtPayment += _profit;
             _profit = 0;
         }
+
+        if (toggles.checkDebtOutstanding) {
+            // final check to make sure accounting is correct
+            require(_debtOutstanding == _debtPayment.add(_loss));
+        }
```

Comment:
This is in the wrong method, it should be in `liquidatePosition`, not prepare return